### PR TITLE
Issue673 Optimize removeElement method in doctrine entities

### PR DESCRIPTION
**Pull request by mhabibi**
Fixes https://github.com/symfony/maker-bundle/issues/673

**ManyToMany relations:**
_Before_:
ManyToMany-master
_After_:
ManyToMany-fix

**OneToMany relations:**
_Before_:
OneToMany-master
_After_:
OneToMany-fix

### CODE REVIEW

File: tests/Util/fixtures/add_many_to_many_relation/User_simple_inverse.php [hunk 1]
```diff
@@ -53,10 +53,8 @@ public function addRecipe(Recipe $recipe): self
 
     public function removeRecipe(Recipe $recipe): self
     {
-        if ($this->recipes->contains($recipe)) {
-            $this->recipes->removeElement($recipe);
-            $recipe->removeFood($this);
-        }
+        $this->recipes->removeElement($recipe);
+        $recipe->removeFood($this);
```
Comment (dmaicher) [inline tests/Util/fixtures/add_many_to_many_relation/User_simple_inverse.php:57]: I think to mirror the behavior from before this should be

```php
if ($this->recipes->removeElement($recipe)) {
    $recipe->removeFood($this);
}
```

instead?

**Review by dmaicher — COMMENTED** [review]
nice :+1:

File: tests/Util/fixtures/add_many_to_many_relation/User_simple_inverse.php (hunk 1 above)
Comment (mhabibi) [reply tests/Util/fixtures/add_many_to_many_relation/User_simple_inverse.php]: Fixed, thanks

**Review by dmaicher — APPROVED** [review]
LGTM

**Comment by weaverryan** [comment]
This is very cool. Thank you @mhabibi!

**Pull request merged**